packages/financial-templates-lib/src/price-feed/utils.js

```diff
@@ -36,36 +37,41 @@ exports.BlockHistory = (getBlock, blocks = []) => {
   function getClosestBefore(timestamp) {
     const index = lodash.sortedIndexBy(blocks, { timestamp }, "timestamp");
     // need to check for an exact match in this case, otherwise go to the older block
-    if (blocks[index].timestamp === timestamp) return blocks[index];
+    if (blocks[index] && blocks[index].timestamp === timestamp) return blocks[index];
     return blocks[index - 1];
   }
 
-  // Call to update cache with blocks between sometime in the past from a block number
-  async function fetchBetween(endTime, blockNumber) {
-    if (blockNumber < 0) return blocks;
-    // Block number is optional, if not specified will default to latest block number.
-    const block = await getBlock(blockNumber);
-    assert(block, "Provider returned empty block");
-    if (block.timestamp < endTime) return blocks;
-    blockNumber = block.number;
-    if (has(blockNumber)) return fetchBetween(endTime, blockNumber - 1);
-    insert(block);
-    return fetchBetween(endTime, blockNumber - 1);
-  }
-
   // Main call to update cache, will take care of fetching all blocks, caching and pruning cache.
-  async function update(age, now, startBlock) {
-    assert(age >= 0, "requires age in seconds");
+  async function update(lookback, now) {
```

Comment:
Would like to get @daywiss's eyes on some of these deeper changes to these utils classes.